lib/jxl/enc_cluster.cc

```diff
@@ -77,38 +77,53 @@ float HistogramDistance(const Histogram& a, const Histogram& b) {
 
 // First step of a k-means clustering with a fancy distance metric.
 void FastClusterHistograms(const std::vector<Histogram>& in,
-                           const size_t num_contexts, size_t max_histograms,
+                           const size_t /*num_contexts*/, size_t max_histograms,
                            float min_distance, std::vector<Histogram>* out,
                            std::vector<uint32_t>* histogram_symbols) {
   PROFILER_FUNC;
   size_t largest_idx = 0;
-  for (size_t i = 0; i < num_contexts; i++) {
+  std::vector<uint32_t> nonempty_histograms;
+  nonempty_histograms.reserve(in.size());
+  for (size_t i = 0; i < in.size(); i++) {
+    if (in[i].total_count_ == 0) continue;
     HistogramEntropy(in[i]);
     if (in[i].total_count_ > in[largest_idx].total_count_) {
       largest_idx = i;
```

Comment:
Set it to nonempty_histograms.size() so it immediately has the index for the right vector? Have to maintain the maxcount too then but I don't like that std::find below so much...

Reply:
Eh, I had this originally but it messed up things since I couldn't really do in[largest_idx] reliably... I prefer this version :)